### PR TITLE
mhlo: migrate from MLIR-HLO to StableHlo

### PR DESCRIPTION
**Pull request by ashay**
This patch replaces the ONNX-to-MHLO conversion pipeline with the
ONNX-to-StableHlo conversion pipeline.  Aside from the relatively few
changes to the C++ source files, most of the changes in this patch
update the lit tests, all of which were updated using the
utils/mlir2FileCheck.py script.

This patch also makes this repository point to the StableHLO submodule
(and drops the MLIR-HLO submodule) since our MHLO commit was old and
contained build bugs that were fixed upstream in the StableHLO
repository.

During the migration, I noticed that MatMul operations with dynamic             
dimensions are not lowered correctly.  Specifically, the lowering for           
matmul operations includes StableHLO's BroadcastInDimOp, which requires         
that its dimensions be static.  Fixing this requires a slightly more            
involved fix, but I will file an issue for this and fixing it in a              
separate PR.  For now, I have inserted a guard condition that skips the         
lowering for matmul with dynamic dimensions, so that the pass doesn't           
crash.

### CODE REVIEW

**Comment by philass** [comment]
@jenkins-droid test this please

File: src/Conversion/ONNXToStableHlo/ONNXToStableHloCommon.cpp [hunk 1]
```diff
@@ -2,18 +2,19 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-//====----- ONNXToMhloCommon.cpp - ONNX dialects to Mhlo lowering ---------===//
+//====----- ONNXToStableHloCommon.cpp - ONNX dialects to StableHlo lowering
+//---------===//
```
Comment (AlexandreEichenberger) [inline src/Conversion/ONNXToStableHlo/ONNXToStableHloCommon.cpp:6]: nit formatting; this happens also elsewhere, please do a sweep

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM, approving this PR as it is a step in the right direction (Stable HLO). 

Please let's continue our discussion about HLO involvement in this project and the level of support, either in the #2567 issue and preferably in a weekly onnx-mlir meeting.

File: src/Conversion/ONNXToStableHlo/ONNXToStableHloCommon.cpp (hunk 1 above)
Comment (ashay) [reply src/Conversion/ONNXToStableHlo/ONNXToStableHloCommon.cpp]: Thanks!  Fixed in the most recent commit.

**Comment by sorenlassen** [comment]
@gongsu832 can you please make the jenkins pipelines run? they don't seem to respond to the jenkins-droid command

**Comment by AlexandreEichenberger** [comment]
@jenkins-droid test this please

**Comment by ashay** [comment]
I imagine that this has come up before, so pardon my ignore when I ask, would y'all be open to:
1. letting builds run on a push to branch, instead of requiring one of the admins to manually trigger builds, and
2. enabling ccache/sccache in CI builds?

**Comment by tungld** [comment]
@jenkins-droid test this please

**Pull request merged**